### PR TITLE
Update for RHEL9 support with specific handling of masquerade #181

### PR DESCRIPTION
**Pull request by prb112**
Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: modules/1_bastion/bastion.tf [hunk 1]
```diff
@@ -298,12 +298,28 @@ resource "null_resource" "bastion_packages" {
...
-      "sudo yum install -y wget iptables jq git net-tools vim python3 tar",
-      "iptables -A FORWARD -i env3 -j ACCEPT",
-      "iptables -A FORWARD -o env3 -j ACCEPT",
-      "iptables -t nat -A POSTROUTING -o env2 -j MASQUERADE",
-      "sysctl -w net.ipv4.ip_forward=1"
+      <<EOF
+      sudo yum install -y wget jq git net-tools vim python3 tar firewalld iptables
+
+      os_ver=$(cat /etc/os-release | egrep "^VERSION_ID=" | awk -F'"' '{print $2}')
+      if [[ $os_ver != "9"* ]]
```
Comment (prb112) [inline modules/1_bastion/bastion.tf:305]:
```suggestion
      if [[ $os_ver == "9"* ]]
```

File: modules/2_nbde/files/powervs-setup.yml [hunk 2]
```diff
@@ -26,28 +27,31 @@
     - name: Setup Proxy Info
       become_user: root
       shell: |
-        echo "Setting up proxy details..."
```
Comment (prb112) [inline modules/2_nbde/files/powervs-setup.yml:29]: @gauravpbankar @aditijadhav38 here is the guard

File: modules/1_bastion/bastion.tf [hunk 3]
```diff
@@ -298,12 +298,28 @@ resource "null_resource" "bastion_packages" {
...
-      "sudo yum install -y wget iptables jq git net-tools vim python3 tar",
-      "iptables -A FORWARD -i env3 -j ACCEPT",
-      "iptables -A FORWARD -o env3 -j ACCEPT",
-      "iptables -t nat -A POSTROUTING -o env2 -j MASQUERADE",
-      "sysctl -w net.ipv4.ip_forward=1"
+      <<EOF
+      sudo yum install -y wget jq git net-tools vim python3 tar firewalld iptables
+
+      os_ver=$(cat /etc/os-release | egrep "^VERSION_ID=" | awk -F'"' '{print $2}')
+      if [[ $os_ver == "9"* ]]
```
[the remark this answers is not in the document]
Comment (aditijadhav38) [reply modules/1_bastion/bastion.tf:305]: @prb112 LGTM

File: modules/2_nbde/files/powervs-setup.yml (hunk 2 above)
Comment (gauravpbankar) [reply modules/2_nbde/files/powervs-setup.yml:29]: ok

**Review by gauravpbankar — APPROVED** [review]

**Review by aditijadhav38 — APPROVED** [review]
LGTM

**Pull request merged**